warehouse-management-service/pkg/database/postgres/shelf.go

```diff
@@ -0,0 +1,198 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	wms "warehouse-management-service"
+)
+
+type shelfQueries interface {
+	createShelfTx(ctx context.Context, tx *sql.Tx, shelf wms.Shelf) error
+	getShelfByIdTx(ctx context.Context, tx *sql.Tx, id string) (wms.Shelf, error)
+	updateShelfTx(ctx context.Context, tx *sql.Tx, shelf wms.Shelf) error
+	deleteShelfTx(ctx context.Context, tx *sql.Tx, id string) error
+	shelfBlockExistsTx(ctx context.Context, tx *sql.Tx, id string) (bool, error)
+}
+
+type shelfQueriesImpl struct{}
+
+type ShelfService struct {
+	queries shelfQueries
+	db      *sql.DB
+}
+
+func NewShelfService(db *sql.DB) *ShelfService {
+	return &ShelfService{
+		queries: new(shelfQueriesImpl),
+		db:      db,
+	}
+}
+
+var InvalidShelfBlock = errors.New("invalid shelfBlockId")
+
+func (s *ShelfService) GetShelfById(ctx context.Context, id string) (wms.Shelf, error) {
+	tx, err := s.db.BeginTx(ctx, nil)
+	if err != nil {
+		return wms.Shelf{}, err
+	}
+
+	shelf, err := s.queries.getShelfByIdTx(ctx, tx, id)
+	switch err {
+	case nil:
+		return shelf, tx.Commit()
+	case sql.ErrNoRows:
+		return wms.Shelf{}, wms.ShelfDoesNotExist
+	default:
+		return wms.Shelf{}, err
+	}
+}
+
+func (s *ShelfService) CreateShelf(ctx context.Context, shelf wms.Shelf) error {
+	tx, err := s.db.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	err = s.queries.createShelfTx(ctx, tx, shelf)
+	switch err {
+	case InvalidShelfBlock:
+		return wms.InvalidShelfBlock
+	case nil:
+		return tx.Commit()
+	default:
+		return err
+	}
+}
+
+func (s *ShelfService) UpdateShelf(ctx context.Context, shelf wms.Shelf) error {
+	tx, err := s.db.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	err = s.queries.updateShelfTx(ctx, tx, shelf)
+	switch err {
+	case nil:
+		return tx.Commit()
+	case RowDoesNotExist:
+		return wms.ShelfDoesNotExist
+	case InvalidShelfBlock:
+		return wms.InvalidShelfBlock
+	default:
+		return err
+	}
+}
+
+func (s *ShelfService) DeleteShelfById(ctx context.Context, id string) error {
+	tx, err := s.db.BeginTx(ctx, nil)
```

Comment:
I think all CRUD operations can be moved from transactions to a single query.

Reply:
I'm slightly confused since you reacted 👍 to above comment and I still see transactions being used.
If you intend to keep them, please resolve this conversation.

Reply:
I intend to refactor this once the CI/CD story I am working on currently is done, for the time being not changing this.